proto/substrait/algebra.proto

```diff
@@ -845,6 +848,11 @@ message Expression {
       int32 scale = 3;
     }
 
+    message Timestamp {
+      // the maximum precision is 9.
+      int32 precision = 1;
```

Comment:
Do we also have a value?

Reply:
Not exactly sure why value would be necessary?

Reply:
This is a literal (the type is named `ParameterizedPrecisionTimestamp`?.  In other words, if my SQL filter is `event_time < '1999-01-08 04:05:06'` and `event_time` is a "precision timestamp" column then I need to make a Substrait plan that looks like...

`less_than(input_field(0), ???)` where `???` is my timestamp literal.  So we do need a value (and not a precision) here.

We could do something like:

```
      // If the precision is 6 or less then this is the microseconds since the UNIX epoch
      // If the precision is more than 6 then this is the nanoseconds since the UNIX epoch
      uint64 precision_timestamp = 34;
```